Enhance description for data asset creation

vivek-dani opened the pull request:
1) Explained data asset creation and updation for vision jobs.
2) Provided example on how to create data asset from data in Azure Blob Storage

vivek-dani committed:
Enhance description for data asset creation

1) Explained data asset creation and updation for vision jobs.
2) Provided example on how to create data asset from data in Azure Blob Storage

vivek-dani committed:
Improved explanation

Made the explanation simpler and more verbose

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -50,6 +50,10 @@ If you already have a data labeling project and you want to use that data, you c
 ### Using pre-labeled training data
```
Need similar changes for v1 version of the doc as well.

vivek-dani replied:
I'll create a seperate PR for v1

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -50,6 +50,10 @@ If you already have a data labeling project and you want to use that data, you c
 ### Using pre-labeled training data
 If you have previously labeled data that you would like to use to train your model, you will first need to upload the images to the default Azure Blob Storage of your Azure ML Workspace and register it as a data asset. 
 
+Below script uploads the image data at path "./data/odFridgeObjects" on your local machine to Azure Blob Storage. Thereafter, it creates a new data asset with the name "fridge-items-images-object-detection" in your Azure ML Workspace. Datastore property of this newly created data asset is made to point to location in Azure Blob Storage where the image data was uploaded. 
```
Nit: Below code snippet instead of "Below scripts"

vadthyavath replied:
minor:  Below script uploads the image data at path "./data/odFridgeObjects" on your local machine to Azure Blob Storage. --> The below script uploads the image data on your local machine at path "./data/odFridgeObjects" to Azure Blob Storage.

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -50,6 +50,10 @@ If you already have a data labeling project and you want to use that data, you c
 ### Using pre-labeled training data
 If you have previously labeled data that you would like to use to train your model, you will first need to upload the images to the default Azure Blob Storage of your Azure ML Workspace and register it as a data asset. 
 
+Below script uploads the image data at path "./data/odFridgeObjects" on your local machine to Azure Blob Storage. Thereafter, it creates a new data asset with the name "fridge-items-images-object-detection" in your Azure ML Workspace. Datastore property of this newly created data asset is made to point to location in Azure Blob Storage where the image data was uploaded. 
```
Could you please clarify what this line means

vivek-dani replied:
Trying to explain 2 steps
1) upload images to blob
2) create data asset from blob
Suggestions for better way of phrasing it are welcome :)

PhaniShekhar replied:
Thanks. The current changes look good.

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,6 +80,35 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 [!Notebook-python[] (~/azureml-examples-main/sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb?name=upload-data)]
 ---
 
+If you already have your data present in Azure Blob Storage and want to create data asset out of it, you can do so by providing path to the location in Azure Blob Storage as shown below.
```
Is it possible to just use the path on the azure blob storage in our jsonl files instead of creating a data asset.

vivek-dani replied:
It appears creating data asset is not mandatory. Why do we have the section to create data asset in our notebooks ?
If we remove the use of data asset, then we should remove it from notebooks as well, before we remove it from docs. 
If we only skip reference to data asset while explaining direct data access from blob, then it changes rest of the code in notebook, which may create confusion

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,6 +80,35 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 ---
 
+If you already have your data present in Azure Blob Storage and want to create data asset out of it, you can do so by providing path to the location in Azure Blob Storage as shown below.
+
```
Should the below code snippets have code to create the data asset - "az ml data create" for CLI and "create_or_update" for SDK ?

vivek-dani replied:
I thought it'd be just repetition. Hence, was thinking to only  highlight the change. Do you think, we should write the whole code snippet again ?

PhaniShekhar replied:
It may not be very clear to the reader, that they have to create a data asset again. Maybe add a note that they have to do this and point to above section.

vivek-dani replied:
done

PhaniShekhar replied:
Thanks Vivek. The changes look good now :)

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,6 +80,35 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
+$schema: https://azuremlschemas.azureedge.net/latest/data.schema.json
+name: fridge-items-images-object-detection
+description: Fridge-items images Object detection
+path: azureml://subscriptions/<my-subscription-id>/resourcegroups/<my-resource-group>/workspaces/<my-workspace>/datastores/<my-datastore>/paths/<path_to_image>
```
Please make sure that this works. Also, nit: <path_to_data_folder> instead of <path_to_image>

vivek-dani replied:
I have tested this

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,6 +80,35 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 [!Notebook-python[] (~/azureml-examples-main/sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb?name=upload-data)]
 ---
 
+If you already have your data present in Azure Blob Storage and want to create data asset out of it, you can do so by providing path to the location in Azure Blob Storage as shown below.
```
I think we should also content here on how to link an existing "Azure blob storage" into a user's AML workspace. The current changes assume that the blob storage is already registered as a datastore in AML workspace.

vivek-dani replied:
Direct blob store path didnt work for me. I tried something like https://vidaniwrk5973817499.blob.core.windows.net/azureml-blobstore-7a82bcb5-09ec-47c5-a392-54051cc33f63/LocalUpload/b1cdf8036e04ed75d810c9fe7cce2f8d/odFridgeObjects",  it didn't work

vadthyavath commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -50,6 +50,10 @@ If you already have a data labeling project and you want to use that data, you c
 
+Below script uploads the image data at path "./data/odFridgeObjects" on your local machine to Azure Blob Storage. Thereafter, it creates a new data asset with the name "fridge-items-images-object-detection" in your Azure ML Workspace. Datastore property of this newly created data asset is made to point to location in Azure Blob Storage where the image data was uploaded. 
+
+If there already exists a data asset with name "fridge-items-images-object-detection" in your Azure ML Workspace, then it'll update its version number and update its datastore property to point to new location in Azure Blob Storage where we uploaded the image data.
```
update its version number and update its datastore property - >update its version number and datastore property?

vivek-dani replied:
The later part gets continued as "update its datastore property to point to new location in Azure Blob Storage". If we do 
"update its version number and datastore property to point to new location in Azure Blob Storage". It'd imply version number would also point to new location in Azure blob storage ?

Above was my understanding. But, please let me know your perspective, I am more than happy to correct it

vadthyavath commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
````diff
@@ -76,6 +80,35 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
+ [!INCLUDE [sdk v2](../../includes/machine-learning-sdk-v2.md)]
+ 
+```Python
+my_data = Data(
````
just curious, do we need to mention imports here? Data() is not very clear!

vivek-dani committed:
Remove redundant imports of notebooks

1) Remove redundant imports of notebooks
2) Re-phrased explanation

vivek-dani committed:
Corrected the explanation of datastore

vivek-dani committed:
create how-to-prepare-datasets-for-automl-images folder

vivek-dani committed:
add update-datastore-authentication.png

Added image to show how to update authentication details in datastore

vivek-dani committed:
Deleted .gitkeep

vivek-dani committed:
Use existing Blob storage as datastore

Provide description on how to use existing Azure Blob storage as datastore in AML workspace

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,10 +80,44 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 
+
+### Using pre-labeled training data from Azure Blob storage
+If you have your labelled training data present in a container in Azure Blob storage, then you can access it directly from there by [creating a datastore referring to that container](how-to-prepare-datasets-for-automl-images.md#create-an-azure-blob-datastore). Once you have created a datastore in AML workspace, linked to a existing container in blob, you'll have to update authentication details f ...
```
The link is broken here.

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,10 +80,44 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 
+
+### Using pre-labeled training data from Azure Blob storage
+If you have your labelled training data present in a container in Azure Blob storage, then you can access it directly from there by [creating a datastore referring to that container](how-to-prepare-datasets-for-automl-images.md#create-an-azure-blob-datastore). Once you have created a datastore in AML workspace, linked to a existing container in blob, you'll have to update authentication details f ...
```
Why do we need to do this step here ? 
I would suggest to remove this comment on authentication details and the screenshot below for the following reasons:
- Ideally this information should be in the "registering a datastore" page. 
- If you create a datastore using "account key" or "sas", this step is not needed. 
- Authentication with data might be a more complicated topic and there is some documentation around it here: https://learn.microsoft.com/en-us/azure/machine-learning/how-to-administrate-data-authentication We may not cover all the scenarios with the document we currently have.

vivek-dani replied:
I agree that we should remove it from here. But I think it should be specified somewhere. There is no clear info/screentshots anywhere which shows steps to register blob as datastore.  Should I later create a separate PR with authentication info [here](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-datastore?tabs=cli-identity-based-access%2Ccli-adls-identity-based-access%2Ccli-azfiles-account-key%2Ccli-adlsgen1-identity-based-access)  ?

PhaniShekhar replied:
I think this should be owned by data team. Could you please reach out to Qianqian Nie and suggest to improve this part of their documentation.

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -76,10 +80,44 @@ az ml data create -f [PATH_TO_YML_FILE] --workspace-name [YOUR_AZURE_WORKSPACE]
 [!Notebook-python[] (~/azureml-examples-main/sdk/jobs/automl-standalone-jobs/automl-image-object-detection-task-fridge-items/automl-image-object-detection-task-fridge-items.ipynb?name=upload-data)]
 ---
 
+If you already have your data present in an existing datastore and want to create data asset out of it, you can do so by providing path to the data in datastore as shown below, instead of providing path on your local machine.
```
Do we need to do this if data is already in datastore ? Would using that path directly in the jsonl file not work ?

vivek-dani replied:
here's my understanding/observation: I directly use path to datastore directly in jsonl, then it creates a new-different-data assets at runtime, for every single run. So, for n runs it'll create n different data assets, creating redundancy and messiness in data asset catalog for user. Hence, there is some benefit I see in creating data asset, even when datastore path can be directly used......What do u think ?

PhaniShekhar replied:
Ohh I see. It makes sense to have this section then. Thanks for the explanation

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
```diff
@@ -47,9 +47,13 @@ It helps to create, manage, and monitor data labeling tasks for
 
+Below scripts uploads the image data on your local machine at path "./data/odFridgeObjects" to datastore in Azure Blob Storage. Thereafter, it creates a new data asset with the name "fridge-items-images-object-detection" in your Azure ML Workspace. 
+
+If there already exists a data asset with name "fridge-items-images-object-detection" in your Azure ML Workspace, then it'll update its version number of data asset and make it point to new location in datastore in Azure Blob Storage where we uploaded the image data.
```
Nit:
- If there is already a data asset ....
- then it will update version number of the data asset and point it to the new location where image data is uploaded.

vivek-dani committed:
Removed datastore authentication part

vivek-dani committed:
Delete articles/machine-learning/media/how-to-prepare-datasets-for-automl-images directory

delete dir azure-docs/articles/machine-learning/media/how-to-prepare-datasets-for-automl-images/

vivek-dani committed:
Update data asset with new datastore location

Added pointer to above section in section where we explain how we can update existing data asset with new datastore location

vivek-dani committed:
Merge branch 'main' into patch-5

PhaniShekhar commented on articles/machine-learning/how-to-prepare-datasets-for-automl-images.md:
````diff
@@ -47,36 +47,12 @@ It helps to create, manage, and monitor data labeling tasks for
-[!INCLUDE [cli v2](../../includes/machine-learning-cli-v2.md)]
+Below scripts uploads the image data on your local machine at path "./data/odFridgeObjects" to datastore in Azure Blob Storage. Thereafter, it creates a new data asset with the name "fridge-items-images-object-detection" in your Azure ML Workspace. 
 
-```yaml
````
This seems to be a bad merge. We should not remove the content about using datasets from labeling. Please fix this.

vivek-dani replied:
I am not sure, if I understood correctly. I have only renamed the title of section to " Using pre-labeled training data from local machine". Original para below it is as-is

vivek-dani committed:
Added code snippet related to Data Labeling tool 

resolved merge conflict

jborsecnik committed:
Update how-to-prepare-datasets-for-automl-images.md
